### PR TITLE
add debugging documentation

### PR DESCRIPTION
**Pull request by astoycos**
Add information on how to debug bpfd with vscode.

### CODE REVIEW

File: docs/developer-guide/debugging.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                "kind": "bin"
+            }
+        },
+        "cwd": "${workspaceFolder}",
+    },
+    ``` 
+3. On the VM or Server install `lldb-server`
+    ```bash
+        sudo apt install lldb-server
+    ```
```
Comment (Billy99) [inline docs/developer-guide/debugging.md:36]:
```suggestion
    `dnf` based OS:

    ```bash
        sudo dnf install lldb-server
    ```

    `apt` based OS:

    ```bash
        sudo apt install lldb-server
    ```
```

However, on Fedora:
```
$ sudo dnf install lldb-server
Last metadata expiration check: 3:03:50 ago on Wed 04 Oct 2023 04:08:54 AM EDT.
No match for argument: lldb-server
Error: Unable to find a match: lldb-server
```
So what is the equivalent, just `sudo dnf instal lldb`?

File: docs/developer-guide/debugging.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Debugging using VSCode and lldb on a remote machine or VM 
+
+1. Install [code-lldb vscode extension](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb)
+2. Add a configuration to `.vscode/launch.json` like the following:
```
Comment (Billy99) [inline docs/developer-guide/debugging.md:4]:
```suggestion
2. Add a configuration to `.vscode/launch.json` like the following (customizing for a given system using the comment in the configuration file):
```

**Review by Billy99 — COMMENTED** [review]
Thanks. This will be useful!

File: docs/developer-guide/debugging.md (hunk 1 above)
Comment (astoycos) [reply docs/developer-guide/debugging.md:36]: Yeah it's actually included in lldb I believe

**Review by Billy99 — APPROVED** [review]
LGTM

**Pull request merged**